x-pack/plugins/fleet/server/services/epm/archive/parse.ts

```diff
@@ -201,7 +201,10 @@ export function parseAndVerifyArchive(
 ): ArchivePackage {
   // The top-level directory must match pkgName-pkgVersion, and no other top-level files or directories may be present
   const logger = appContextService.getLogger();
-  const toplevelDir = topLevelDirOverride || paths[0].split('/')[0];
+  let toplevelDir = topLevelDirOverride || '';
```

Comment:
This fixed my issue. But since I am not aware of the Fleet/Integrations ecosystem, it might not be the correct way to fix the root of the problem.

Reply:
the fix looks good to me, if you make the pr ready for review, the CI build will run automatically

Reply:
I checked this fix locally and it solved the related error 👍🏻

Reply:
I opened it up for review!